subsys/net/lib/coap/coap.c

```diff
@@ -49,53 +49,132 @@ LOG_MODULE_REGISTER(net_coap, CONFIG_COAP_LOG_LEVEL);
 /* The CoAP message ID that is incremented each time coap_next_id() is called. */
 static uint16_t message_id;
 
-static inline bool append_u8(struct coap_packet *cpkt, uint8_t data)
+static int parse_option(uint8_t *data, uint16_t offset, uint16_t *pos, uint16_t max_len,
+			uint16_t *opt_delta, uint16_t *opt_len, struct coap_option *option);
+static int insert_option(struct coap_packet *cpkt, uint16_t code, const uint8_t *value,
+			 uint16_t len);
+
+#define INSERT_U8(cpkt, offst, data)                                                               \
```

Comment:
I'm not really a fan of the name collision here, there already is a function called `insert_u8()`. Perhaps we could call it `encode_u8()`? And I'd prefer to make it a regular static inline function instead of a macro. Same for the macros below.